### PR TITLE
Added Dockerfile and Docker Compose configuration file

### LINKED ISSUE #26: Update Docker Compose configuration file
The [Docker Compose file](https://github.com/creativecommons/vocabulary/blob/master/Dockerfile) for the project, written in the initial days of the project has now gone out of sync with the capabilities and features of the project. The goal is to update the file in accordance with the current state and eventually push for it to the the standard development environment for the project.

### PR DESCRIPTION
**Pull request by krhitesh**
## Fixes

Fixes #26  by @dhruvkb 

## Description

This PR addes the Dockerfile and docker compose file in accordance with the current state and eventually push for it to the the standard development environment for the project.

Also documentation built via `npm run build:documentation` is not served currently.

## Tests
```
# Build the docker image
docker build . -t ccvocabulary:latest

# Run via docker compose
docker-compose up
```
Visit `localhost:8080` to test.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by krhitesh** [comment]
@dhruvkb Please take a look.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM node:12.16.1-alpine
```
Comment (dhruvkb) [inline Dockerfile]: You can just use the [Dockerfile from `creativecommons/vue-vocabulary`](https://github.com/creativecommons/vue-vocabulary/blob/master/Dockerfile).

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+##############################
+# Development Docker Compose #
+##############################
+
+# This is the version of the Docker Compose standard
+version: '3.4'
+
+services:
+  # The website is the landing page for the CC Vocabulary project
+  website:
```
Comment (dhruvkb) [inline docker-compose.yml]: We're serving the Storybook so that would be a better name for the service.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    # Use the CC Vocabulary image
+    image: ccvocabulary:latest
```
Comment (dhruvkb) [inline docker-compose.yml]: Add a `restart` key.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    # The command to run in the container
+    command: npm run build:documentation
+
+    # Volumes attached to the container
+    volumes:
+      # Enable code changes to reflect in the container
+      - .:/codebase
+      # Ensures that host node_modules are not mounted
+      - node_modules:/codebase/node_modules
```
Comment (dhruvkb) [inline docker-compose.yml]: One-off jobs like building should not be performed using Docker Compose. Docker Compose is for running persistent services.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
You missing running the SCSS compilation service `npm run watch` in conjuction with Storybook in `docker-compose.yml`.

File: docker-compose.yml (hunk 3 above)
Comment (dhruvkb) [inline docker-compose.yml]: Name the image "vocabulary".

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 4 above)
Comment (dhruvkb) [reply docker-compose.yml]: You can write a shell script that spins off a container with the "vocabulary" image and runs the build process inside it, terminating the container when done.

Comment (krhitesh) [reply docker-compose.yml]: Is there a need to run `build:documentation` within the container?

Comment (krhitesh) [reply docker-compose.yml]: > You missing running the SCSS compilation service `npm run watch` in conjuction with Storybook in `docker-compose.yml`.

Taking a look at `docker-compose.yml` in [vue-vocabulary](https://github.com/creativecommons/vue-vocabulary), I think `npm run storybook` also watches for file changes as well. Isn't it?

Comment (krhitesh) [reply docker-compose.yml]: > Is there a need to run `build:documentation` within the container?

I could just put `npm run build:documentation` inside Dockerfile in that case.

Comment (dhruvkb) [reply docker-compose.yml]: In Vue Vocabulary, Storybook autoreloads when styles change. In Vocabulary, it doesn't. So you'll have to run the watch service alongside.

If you put `npm run build:documentation` inside the container, won't that build everytime we bring up the container?

Comment (krhitesh) [reply docker-compose.yml]: > You can write a shell script that spins off a container with the "vocabulary" image and runs the build process inside it, terminating the container when done.

Sure I can do that but is it necessary to use `npm run build: documentation` in docker? What's its significance in docker if we are serving via storybook and watch command?

**Comment by krhitesh** [comment]
@dhruvkb Please take a look

**Review by dhruvkb — APPROVED** [review]

**Comment by brylie** [comment]
@all-contributors please add @krhitesh for platform, code

**Pull request merged**